beetsplug/parentwork.py

```diff
@@ -0,0 +1,159 @@
+            work_disambig        = []
+            parent_work          = []
+            parent_work_disambig = []
+            parent_composer      = []
+            parent_composer_sort = []
+            item.read()
+            recording_id = item['mb_trackid']
+            performer_types = ['performer', 'instrument', 'vocal',
+                'conductor', 'performing orchestra', 'chorus master', 
+                    'concertmaster']
+            i = 0
+            while i < 5:
+                try:
+                    rec_rels = musicbrainzngs.get_recording_by_id(
+                        recording_id, includes=['work-rels', 'artist-rels'])
+                    if 'artist-relation-list' in rec_rels['recording']:
+                        for dudes in rec_rels['recording'][
+                                'artist-relation-list']:
+                            if dudes['type'] in performer_types:
+                                performer.append(dudes['artist']['name'])
+                                    performer_sort.append(dudes['artist']
+                                        ['sort-name'])
+                    if 'work-relation-list' in rec_rels['recording']:
+                        for work_relation in rec_rels['recording'][
+                                'work-relation-list']:
+                            if work_relation['type'] != 'performance':
+                                continue
+                            work_id = work_relation['work']['id']
+                            work.append(work_relation['work']['title'])
+                            if 'disambiguation' in work_relation['work']:
+                                work_disambig.append(work_relation['work']
+                                    ['disambiguation'])
+                            work_info = find_parentwork(work_id)
+                            if 'artist-relation-list' in work_info['work']:
+                                for artist in work_info['work'][
+                                    'artist-relation-list']:
+                                    if artist['type'] == 'composer' and not\
+                                        artist['artist']['name'] in\
+                                        parent_composer:
+                                        parent_composer.append(artist\
+                                            ['artist']['name'])
+                                        parent_composer_sort.append(
+                                            artist['artist']
+                                            ['sort-name'])
+                            else:
+                                print('no composer')
+                                print('add one at')
+                                print('https://musicbrainz.org/work/' + 
+                                    work_info['work']['id'])
+                            if work_info['work']['title'] in parent_work:
+                                pass
```

Comment:
Right; I agree that storing the IDs themselves isn't clearly useful. But using a Python `set()` of IDs as a filter will make the code a little clearer (and a little more correct).